tools/python/m2k-osc/ain_tab.py

```diff
@@ -0,0 +1,199 @@
+        samplerate_options = tk.OptionMenu(self, in_samplerate,"1000","10000","100000","1000000","10000000","100000000")
+        samplerate_options.grid(row=4,column=1)
+        ain_samples_label=tk.Label(self,text="Samples to read")
+        ain_samples_label.grid(row=5,column=0)
+        global read_samples
+        read_samples=tk.StringVar()
+        read_samples.set("1000")
+        self.ain_samples_entry=tk.Entry(self,textvariable=read_samples)
+        self.ain_samples_entry.grid(row=5,column=1)
+        ain_samples_submit=tk.Button(self,text="OK",command=self.update_readsamples)
+        ain_samples_submit.grid(row=5,column=2)
+        ain_ch0_label = tk.Label(self,text="Set-up for Channel 1")
+        ain_ch0_label.grid(row=6,column=1)
+        ain_ch1_label=tk.Label(self,text="Set-up for Channel 2")
+        ain_ch1_label.grid(row=6,column=2)
+        ain_range_label=tk.Label(self,text="Range")
+        ain_range_label.grid(row=7,column=0)
+        global ain_range_ch0_var,ain_range_ch1_var
+        ain_range_ch0_var= tk.IntVar()
+        ain_range_ch0_check=tk.Checkbutton(self,text ="Set High Range",variable=ain_range_ch0_var,command=self.set_range_ch0)
+        ain_range_ch0_check.grid(row=7,column=1)
+        ain_range_ch1_var= tk.IntVar()
+        ain_range_ch1_check=tk.Checkbutton(self,text ="Set High Range",variable=ain_range_ch1_var,command=self.set_range_ch1)
+        ain_range_ch1_check.grid(row=7,column=2)
+        ain_rangelimits_label=tk.Label(self,text="Range Limits")
+        ain_rangelimits_label.grid(row=8,column=0)
+        global rangelimits_ch0,rangelimits_ch1
+        rangelimits_ch0=tk.StringVar()
+        rangelimits_ch1=tk.StringVar()
+        ain_rangelimits_ch0=tk.Label(self,textvariable=rangelimits_ch0)
+        ain_rangelimits_ch0.grid(row=8,column=1)
+        ain_rangelimits_ch1=tk.Label(self,textvariable=rangelimits_ch1)
+        ain_rangelimits_ch1.grid(row=8,column=2)
+        ain_startch0=tk.Button(self,text="Read",command=self.start_in)
+        ain_startch0.grid(row=9,column=1)
+        ain_stopch0=tk.Button(self,text="Stop",command=self.stop_in)
+        ain_stopch0.grid(row=9,column=2)
+
+    def change_sample_rate(self):
+        desired_samplerate=float(in_samplerate.get())
+        current_samplerate=ain.getSampleRate()
+        ain.setOversamplingRatio(int(current_samplerate/desired_samplerate))
+        print(ain.getOversamplingRatio())
+    def create_canvas(self):
+        x=[1,2,3]
+        y_ch0=[5,4,7]
+        y_ch1=[8,6,5]
+        self.linech0,=fig_subplot.plot(x,y_ch0)
+        self.linech1,=fig_subplot.plot(x,y_ch1)
+        self.canvas=FigureCanvasTkAgg(fig,master=self)
+        self.canvas.draw()
+        self.canvas.get_tk_widget().grid(row=11,column=0,columnspan=4)
+    def enable_ch0(self):
+        self.get_ain()
+        if ain_ch0_var.get() ==1:
+            ain.enableChannel(0,True)
+            print(ain.getRange(0))
```

Comment:
this print looks like debug
intended or forgotten?

Reply:
forgotten debug print. gone now